src/cluster_legacy.c

```diff
@@ -3855,6 +3857,24 @@ int clusterProcessPacket(clusterLink *link) {
                     clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG | CLUSTER_TODO_UPDATE_STATE | CLUSTER_TODO_FSYNC_CONFIG);
                 }
 
+                if (sender_claimed_primary && nodeIsReplica(sender_claimed_primary)) {
+                    if (nodeIsReplica(myself) && areInSameShard(myself, sender_claimed_primary) &&
```

Comment:
Is it possible that the sender_claimed_primary actually has a different shard_id, since we are just learning about this through gossip? In this specific instance the sender_claimed_primary object has our our local view of its shard_id, but it may in reality have a different shard_id.

Also below, `clusterSetNodeAsPrimary` doesn't configure us to do replication, so we'll mark it as our primary but we won't start replicating from it.